File: Thought-works/Round-1-HackerRank/dialpad.py
# Design a dial pad that enters two integers and do the permutation

"""
i/p: 23
abc def 
o/p: "AD AE Af BD BE BF CD CE CF"

"""

s = {}  # empty dictionary

s[2] = "ABC"
s[3] = "DEF"
s[4] = "GHI" 
s[5] = "JKL"
s[6] = "MNO"
s[7] = "PQRS"
s[8] = "TUV"
s[9] = "WXYZ"

num = int(input("Enter a number :"))


def permutation(num):
    res = ""
    if(num == 10):
        print("N/A")
    else:
        for i in range(len(s[int(num/10)])):
            for j in range(len(s[num%10])):
                res = res + s[int(num/10)][i] + s[int((num%10))][j] + " "
    return res

print(permutation(num))

File: Thought-works/Round-1-HackerRank/dialpad.java
public class dialpad {
    static String printPermutations(int n) {
        String arr[] = new String[10];

        // arr[0] = "null";
        // arr[1] =
        arr[2] = "ABC";
        arr[3] = "DEF";
        arr[4] = "GHI";
        arr[5] = "JKL";
        arr[6] = "MNO";
        arr[7] = "PQRS";
        arr[8] = "TUV";
        arr[9] = "WXYZ";

        int x = n / 10; // 2
        int y = n % 10; // 3

        // System.out.println(arr[x].charAt[i])

        String res = "";
        for (int i = 0; i < arr[x].length(); i++) {
            for (int j = 0; j < arr[y].length(); j++) {
                res = res + arr[x].charAt(i) + arr[y].charAt(j) + " ";
            }
        }

        return res;

    }

    public static void main(String[] args) {
        String s = printPermutations(23);
        System.out.println(s);

    }
}
